Common/src/main/java/com/blamejared/crafttweaker/impl/util/sequence/SequenceManager.java

```diff
@@ -0,0 +1,113 @@
+package com.blamejared.crafttweaker.impl.util.sequence;
+
+import com.blamejared.crafttweaker.api.util.sequence.Sequence;
+import com.blamejared.crafttweaker.api.util.sequence.SequenceType;
+
+import java.util.*;
+
+/**
+ * Manages {@link Sequence}s, handles adding, removing and ticking.
+ */
+public class SequenceManager {
+    
+    private static final Map<SequenceType, List<Sequence<?, ?>>> clientSequences = new HashMap<>();
+    private static final Map<SequenceType, List<Sequence<?, ?>>> clientSequencesView = Collections.unmodifiableMap(clientSequences);
+    private static final Map<SequenceType, List<Sequence<?, ?>>> serverSequences = new HashMap<>();
+    private static final Map<SequenceType, List<Sequence<?, ?>>> serverSequencesView = Collections.unmodifiableMap(serverSequences);
```

Comment:
How would a sequenceType Approach work for other types, like PlayerTicks?
Or would each player be its own sequenceType?

Reply:
The type is just used to make sure we tick things correct, like the PlayerTickEvent would tick the PLAYER sequence type